src/redux/features/Plantings/plantingsSelectors.ts

```diff
@@ -1,11 +1,48 @@
 import { RootState } from 'src/redux/rootReducer';
+import { selectPlantingSites } from '../tracking/trackingSelectors';
+import { createSelector } from '@reduxjs/toolkit';
+import { PlantingSearchData } from './plantingsSlice';
 
 export const selectPlantings = (state: RootState) => state.plantings?.plantings;
 
+export const selectTotalPlantsBySubzone = (state: RootState) => {
+  const plantingsBySubzone: Record<number, { totalPlants: number }> = {};
+  state.plantings?.plantings?.forEach((planting) => {
+    if (plantingsBySubzone[Number(planting.plantingSubzone.id)]) {
+      plantingsBySubzone[Number(planting.plantingSubzone.id)] = {
+        totalPlants: plantingsBySubzone[Number(planting.plantingSubzone.id)].totalPlants + planting.totalPlants,
+      };
+    } else {
+      plantingsBySubzone[Number(planting.plantingSubzone.id)] = { totalPlants: planting.totalPlants };
+    }
+  });
+  return plantingsBySubzone;
+};
+
 export const selectPlantingsDateRange = (state: RootState, dateRange: string[]) =>
   selectPlantings(state)?.filter((planting) => {
     if (!dateRange || dateRange.length === 0) {
       return true;
     }
     return planting.createdTime > dateRange[0] && (dateRange.length < 2 || planting.createdTime < dateRange[1]);
   }) ?? [];
+
+export const selectPlantingProgressSubzones = createSelector(
+  [(state: RootState) => selectPlantingSites(state), (state: RootState) => selectTotalPlantsBySubzone(state)],
+  (plantingSites, plantingsBySubzone) => {
+    plantingSites?.flatMap((ps) => {
```

Comment:
flatmap is fine.. requires going through the entire list when finding or filtering by site id, if you use a map instead of flatmap, you can store it by key/results
{ siteId, siteName, reported: [..] } for easier filter/find.. up to you. pros/cons to either approach